src/hotspot/share/services/memTracker.cpp

```diff
@@ -52,14 +52,14 @@ NMT_TrackingLevel MemTracker::_tracking_level = NMT_unknown;
 
 MemBaseline MemTracker::_baseline;
 
-void MemTracker::initialize() {
+void MemTracker::initialize(ccstr value) {
   bool rc = true;
   assert(_tracking_level == NMT_unknown, "only call once");
 
-  NMT_TrackingLevel level = NMTUtil::parse_tracking_level(NativeMemoryTracking);
+  NMT_TrackingLevel level = NMTUtil::parse_tracking_level(value);
   // Should have been validated before in arguments.cpp
```

Comment:
Why don't we save the value parsed in arguments.cpp so we don't need to repeat the parsing?

Reply:
Fixed.

I moved the relevant NMT related argument parsing into one place, hence we only need to parse the setting once.